weechat-notifications.el

```diff
@@ -111,8 +140,14 @@ Supported actions:
             (append (list (cons notifications-id buffer-ptr))
                     weechat--notifications-id-to-msg)))))
 
+(defun weechat-notifications-libnotify-handler (type &optional sender text _date buffer-ptr)
+  "Notification handler using libnotify."
+  (notify-via-libnotify
+   (weechat-notifications-get-title type sender)
+   (if (eq text nil) nil (xml-escape-string text))))
+
 (add-hook 'weechat-notification-handler-functions
-          #'weechat-notifications-handler)
+          weechat-notification-default-handler)
```

Comment:
Try using `#'weechat-notification-default-handler` here. The `#'` gives you the `var` for the symbol `weechat-notification-default-handler`,which is an indirection so if you change the function the var points to later, the update will be reflected in the hook.